### ISSUE TITLE
`text` uses draw colors too

### ISSUE THREAD

**Issue by zetanumbers** [ZetaNumbers/wasm4-rs#3]

**Comment by IFcoltransG** [comment]
It would be a good idea to expose `DRAW_COLORS` in the `framebuffer` API, rather than just using it inside `Blit`. `rect` and `oval` and so on use `DRAW_COLORS` too.

**Comment by wischi-chr** [comment]
I just ran into the same issue and initially couldn't figure out what went wrong. I ended up just writing a slim safe wrapper that allows me to set the draw color.

In my opinion the safe sprite `blit` wrapper shouldn't mess with `DRAW_COLORS` at all (because it's basically a side effect for all other consequitive drawing calls that are not sprites) and a separate function should be exposed to change it.

**Edit/Update:**
Removing the `DRAW_COLORS` from the sprite would also allow for nice optimizations like that:

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/3d5c9b37-b628-4db1-9f4b-951fe0583a67" />

I even think that was the original intention behind that `DRAW_COLORS` API design.

With the current sprite implementation I'd have to create two sprites to draw the clouds and bushes, even if the are technically the same but just different color indizes.

@FabricatorZayac I just saw in your PR that you added the `DrawIndex` to all the other APIs and I also thought about extending this library, what do you think about actually moving the opposite way and removing `DrawIndex` from `Sprite`?

**Comment by FabricatorZayac** [comment]
So, you're proposing to just make a wrapper around setting `DRAW_COLORS` and just manually call it before every draw call? It would allow for some micro optimizations. I guess if I wanted to associate specific colors with specific objects I could just make another wrapper function for it

**Comment by wischi-chr** [comment]
Yes a bit like that: https://github.com/zetanumbers/wasm4-rs/compare/main...wischi-chr:wasm4-rs:wischi_dev
Maybe an `IndexedSprite` and `IndexedSpriteView` which could hold just a reference to a sprite/view and a `DrawIndex`.
I'm not quite sure and currently trying a few things but I'm not a fan of adding `DrawIndices` to all APIs.

Edit:
Or maybe use different "foreground" and "background" settings as palette indizes and use that for text/rect/etc. rendering only? I think I'll try a few things

**Comment by zetanumbers** [comment]
You know, I have feel for some time that current API is over-engineered. For wasm4 demos I've built, I didn't use this lib to see every opportunity raw interface gives me.

I'm not sure how I would refresh design of this library. Maybe I should even delegate crate ownership to someone else because I lack time to think about this.
